crates/matrix-sdk/src/notification_settings/mod.rs

```diff
@@ -199,6 +200,31 @@ impl NotificationSettings {
             }
         };
 
+        let ruleset = self.rules.read().await.clone().ruleset;
+        for rule_id in rule_ids {
+            // Unstable polls push rule may be missing. Skip the update in case.
```

Comment:
Why fully skip in this case?

The docs for the method don't tell you that this might silently do nothing. We either need to document this or at least do some sensible action in this case.

Reply:
We may still throw a  `NotificationSettingsError::RuleNotFound` error in this case.
But how the client would know which rule failed to be updated (we update multiple push rules now)?
Should we provide the `rule_id` as a part of the `RuleNotFound` error?

Reply:
That sounds like a good addition.

Reply:
Applied here -> https://github.com/matrix-org/matrix-rust-sdk/pull/2644/commits/06dc8356d8079fa6397975ee5ee9be8171dc3f4a